SSA/Core/Framework.lean

```diff
@@ -267,9 +267,48 @@ def Lets.repr (prec : Nat) : Lets d eff Γ t → Format
 instance : Repr (Expr d Γ eff t) := ⟨flip Expr.repr⟩
 instance : Repr (Com d Γ eff t) := ⟨flip Com.repr⟩
 instance : Repr (Lets d Γ eff t) := ⟨flip Lets.repr⟩
-
 end Repr
 
+/- # ToString instances for Com and Expr  -/
+section ToString
+variable {d} [DialectSignature d] [Repr d.Op] [Repr d.Ty][ToString d.Ty][ToString d.Op]
+
+/-- Format a list of formal arguments as `(%0 : t₀, %1 : t₁, ... %n : tₙ)` -/
+partial def formatFormalArgListTupleStr [ToString Ty] (ts : List Ty) : String :=
+  let args := (List.range ts.length).zip ts |>.map
+    (fun (i, t) => s!"%{i} : {toString t}") -- assumes toString function for types. -- tried using it with naming but then difficult to track.
+  "(" ++ String.intercalate ", " args ++ ")"
+
+  partial def Expr.toString1 [ToString d.Op] : Expr d Γ eff t → String
+    | Expr.mk (op : d.Op) _ _ args regArgs =>
+        let outTy : d.Ty  := DialectSignature.outTy op
+        let argTys := DialectSignature.sig op
+
+      s!"{toString op}{formatArgTuple args} : {formatTypeTuple argTys} → ({toString outTy})"
+
+partial def comToStringAux : Com d Γ eff t → String
```

Comment:
Why not:
```suggestion
partial def Com.toStringAux : Com d Γ eff t → String
```
That would be a bit more consistent. You could even go a bit more descriptive and name it `Com.toStringBody` to emphasize it's formatting just the body of the basic block.